pkgs/build-support/trivial-builders/test/references/default.nix

```diff
@@ -89,36 +90,41 @@ let
       mainProgram = "references-test";
     };
   });
+
+  getSelf = lib.getAttrFromPath selfAttrPath;
```

Comment:
Not super happy about this. Could we do `config.node.pkgs.callPackage ./. {}` instead? See next comment.

Reply:
Still a bit messy. Maybe `testScriptBin` should be moved into a file `packages.nix` so that we only construct what we need, making the whole setup less confusing.